Library/Homebrew/cmd/uninstall.rb

```diff
@@ -9,15 +10,41 @@ def uninstall
       ARGV.kegs.each do |keg|
         keg.lock do
           puts "Uninstalling #{keg}... (#{keg.abv})"
+
+          # Find old name and old name symlinks for formula and remove them
+          begin
+            formula = Formulary.from_rack(keg.rack)
```

Comment:
> I need to check the whole Cellar to find that link or somehow find files that are links to bar.

Given there won't be too many symbol links. I think it's acceptable. Part of Homebrew's design choice is that it can work without formula(e.g. we support self build package and use `brew link` etc to manage them). IMO, using file system would be a much robust approach.